code/frameworks/svelte-vite/src/utils.ts

```diff
@@ -14,3 +18,46 @@ export function hasPlugin(plugins: PluginOption[], name: string) {
     })
   );
 }
+
+/**
+ * A migration step that ensures the svelte-vite framework still supports SvelteKit,
+ * but warns the user that they should use the svelte-kit framework instead.
+ * Should be removed when we decide to remove support completely for SvelteKit in svelte-vite
+ */
+export function handleSvelteKit(plugins: PluginOption[]): PluginOption[] {
+  if (!hasPlugin(plugins, 'vite-plugin-svelte-kit')) {
+    // this is not a SvelteKit project ✅
+    return plugins;
+  }
+
+  /*
+  the svelte-kit framework uses this svelte-vite framework under the hood
+  so we have to take extra care of only warning when the user is actually using
+  svelte-vite directly and not just through svelte-kit
+  */
+
+  const packageJsonPath = path.resolve('package.json');
+  if (!fs.existsSync(packageJsonPath)) {
+    // we can't determine if this is using svelte-kit, let's assume it does ✅
+    return plugins;
+  }
+
+  const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'));
+  const allDependencies = [
+    ...Object.keys(packageJson.dependencies || {}),
+    ...Object.keys(packageJson.devDependencies || {}),
+  ];
+  if (
+    allDependencies.includes('@storybook/svelte-kit') &&
+    !allDependencies.includes('@storybook/svelte-vite')
+  ) {
+    // this uses @storybook/svelte-kit, so everything is fine ✅
+    return plugins;
+  }
```

Comment:
I feel this is a little messy, reading the dependencies to determine if we're using the SvelteKit framework or Svelte-Vite. The cleaner way would be to base it on the Storybook config, but I couldn't figure out an easy way to get that in the `viteFinal` call.

Either way, this should all just be temporary code, so I'm not too worried about it. (famous last words)

Reply:
I think you should be able to do `await options.presets.apply('framework')` in `viteFinal`